modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -297,6 +309,13 @@ private void undeploy(T artifact, boolean removeData) {
       deploymentListener.onUndeploymentStart(artifact.getArtifactName());
 
       artifacts.remove(artifact);
+      Optional<Registry> optionalRegistry = ofNullable(artifact.getRegistry());
```

Comment:
no need to use the intermediate Optional just for this. a simple null check will do.How can it be null, anyway?

Reply:
I also thought that it couldn't happen until some tests started failing because getRegistry returned null (deploysDomainWithSharedLibPrecedenceOverApplicationSharedLib, deploysDomainBundleZipAfterStartup, ...) Then I checked getRegistry() and the method implementation indicates that null is a possible return value:
```java
public Registry getRegistry() {
    return artifactContext != null ? artifactContext.getRegistry() : null;
  }
```
 so I ended up adding this additional check.